cmd/wigToTrainingSet/wigToTrainingSet.go

```diff
@@ -0,0 +1,182 @@
+	}
+
+	wigs := wig.GoReadToChan(s.InWigFile)
+	genome := fasta.Read(s.InFastaFile)
+
+	for i := range wigs {
+		chromIndex = getChromIndex(genome, i.Chrom)
+		for start = 0; start < len(i.Values)-s.WindowSize; start += s.Stride {
+			if i.Values[(start+start+s.WindowSize)/2] == s.Missing {
+				continue //skip regions where there is no data in the wig.
+			}
+			currName = fmt.Sprintf("%s:%v-%v", i.Chrom, start, start+s.WindowSize)
+			currFa = fasta.Extract(genome[chromIndex], start, start+s.WindowSize, currName)
+			dna.AllToUpper(currFa.Seq)
+			if s.LogTransform {
+				lineToWrite = fmt.Sprintf("%s\t%s\t%g\n", currFa.Name, dna.BasesToString(currFa.Seq), math.Log(i.Values[(start+start+s.WindowSize)/2]))
+			} else {
+				lineToWrite = fmt.Sprintf("%s\t%s\t%g\n", currFa.Name, dna.BasesToString(currFa.Seq), i.Values[(start+start+s.WindowSize)/2]) //start+window / 2 is the midpoint of the window.
+			}
+			//now we shard the training example into either the testing, training, or validation set.
+			currRand = rand.Float64()
+			if currRand < s.TestingProp {
+				_, err = fmt.Fprintf(testOut, lineToWrite)
+				exception.PanicOnErr(err)
+			} else if currRand < s.TestingProp+s.ValidationProp {
+				_, err = fmt.Fprintf(validateOut, lineToWrite)
+				exception.PanicOnErr(err)
+			} else {
+				_, err = fmt.Fprintf(trainOut, lineToWrite)
+				exception.PanicOnErr(err)
+			}
+
+			if s.IncludeRevComp {
+				dna.ReverseComplement(currFa.Seq)
+				if s.LogTransform {
+					lineToWrite = fmt.Sprintf("%s\t%s\t%g\n", fmt.Sprintf("%s_rev", currFa.Name), dna.BasesToString(currFa.Seq), math.Log(i.Values[(start+start+s.WindowSize)/2]))
+				} else {
+					lineToWrite = fmt.Sprintf("%s\t%s\t%g\n", fmt.Sprintf("%s_rev", currFa.Name), dna.BasesToString(currFa.Seq), i.Values[(start+start+s.WindowSize)/2])
+				}
+				//we don't regenerate currRand, for and rev for same sequence both land in the same shard.
+				if currRand < s.TestingProp {
+					_, err = fmt.Fprintf(testOut, lineToWrite)
+					exception.PanicOnErr(err)
+				} else if currRand < s.TestingProp+s.ValidationProp {
+					_, err = fmt.Fprintf(validateOut, lineToWrite)
+					exception.PanicOnErr(err)
+				} else {
+					_, err = fmt.Fprintf(trainOut, lineToWrite)
+					exception.PanicOnErr(err)
+				}
+			}
+		}
+	}
+	err = trainOut.Close()
+	exception.PanicOnErr(err)
+	err = testOut.Close()
+	exception.PanicOnErr(err)
+	err = validateOut.Close()
+	exception.PanicOnErr(err)
+}
+
+func getChromIndex(genome []fasta.Fasta, chrom string) int {
+	for i := range genome {
+		if genome[i].Name == chrom {
+			return i
+		}
+	}
+	log.Fatalf("Wig chromosome name: %v not found in target genome.", chrom)
+	return -1
+}
+
+func usage() {
+	fmt.Print(
+		"wigToTrainingSet - Converts a wig to a training, validation, and testing set for the GenomeSequenceConvNet.\n" +
+			"Usage:\n" +
+			"wigToTrainingSet input.wig genome.fa train.txt validate.txt test.txt\n" +
+			"options:\n")
+	flag.PrintDefaults()
+}
+
```

Comment:
Can we move this to the top for convention's sake? I was looking for it/confused about Settings